rye/src/cli/publish.rs

```diff
@@ -255,6 +240,22 @@ fn maybe_decrypt(secret: &Secret<String>, yes: bool) -> Result<Secret<String>, E
     bail!("failed to decrypt")
 }
 
+fn secret_from_token(
+    token: String,
+    yes: bool,
+    repo_credentials: &mut Item,
+) -> Result<Option<Secret<String>>, Error> {
+    if token.is_empty() {
+        Ok(None)
```

Comment:
this is the key bit. instead of bailing on the empty string, we return a `None` for the secret